chromadb/test/property/test_persist.py

```diff
@@ -122,6 +123,46 @@ def test_persist(
     del system_2
 
 
+def test_sync_threshold(settings: Settings) -> None:
+    system = System(settings)
+    system.start()
+    client = ClientCreator.from_system(system)
+
+    collection = client.create_collection(
+        name="test", metadata={"hnsw:batch_size": 3, "hnsw:sync_threshold": 3}
+    )
+
+    manager = system.instance(SegmentManager)
+    segment = manager.get_segment(collection.id, VectorReader)
+
+    def get_index_last_modified_at() -> float:
+        try:
+            return os.path.getmtime(segment._get_metadata_file())  # type: ignore[attr-defined]
```

Comment:
this feels terrible... should we update the segment interface to expose this somehow?